src/java/htsjdk/samtools/SAMSequenceRecord.java

```diff
@@ -175,8 +175,15 @@ private URI makeURI(final String s) throws URISyntaxException {
 
     @Override
     public boolean equals(final Object o) {
-        if (this == o) return true;
-        if (!(o instanceof SAMSequenceRecord)) return false;
+        if (this == o) {
+            return true;
+        }
+        if (o == null) {
+            return false;
+        }
+        if (this.getClass() != o.getClass()) {
```

Comment:
Similarly here - I don't think people tend to subclass SAMSequenceRecord, but I think this class should continue to implement logical equality and not require type equality.